plugins/BEdita/Core/src/Filesystem/Thumbnail/GlideGenerator.php

```diff
@@ -0,0 +1,164 @@
+ *
+ * @since 4.0.0
+ */
+class GlideGenerator extends ThumbnailGenerator
+{
+
+    /**
+     * Enforced maximum image size.
+     *
+     * @var int
+     */
+    const MAX_IMAGE_SIZE = 1 << 22; // 2048 * 2048 === 2^11 * 2^11 === 2^22
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $_defaultConfig = [
+        'cache' => 'thumbnails',
+        'driver' => 'gd',
+    ];
+
+    /**
+     * Get destination path for a thumbnail.
+     *
+     * @param \BEdita\Core\Model\Entity\Stream $stream Stream entity instance.
+     * @param array $options Thumbnail options.
+     * @return string
+     */
+    protected function getFilename(Stream $stream, array $options = [])
+    {
+        $ext = pathinfo($stream->file_name, PATHINFO_EXTENSION);
+        if ($ext) {
+            $ext = '.' . $ext;
+        }
+        $filesystem = $this->getConfig('cache', 'thumbnails');
+        $base = $stream->filesystemPath($filesystem);
+        $options = sha1(serialize($options));
+
+        return sprintf(
+            '%s/%s%s',
+            $base,
+            $options,
+            (string)$ext
+        );
+    }
+
+    /**
+     * Get Glide API runner.
+     *
+     * @return \League\Glide\Api\Api
+     */
+    protected function getGlideApi()
+    {
+        $driver = $this->getConfig('driver', 'gd');
+        $api = new GlideApi(
+            new ImageManager(compact('driver')),
+            [
+                new OrientationManipulator(),
+                new CropManipulator(),
+                new SizeManipulator(static::MAX_IMAGE_SIZE),
+                new BlurManipulator(),
+                new EncodeManipulator(),
+            ]
+        );
+
+        return $api;
+    }
+
+    /**
+     * Generate a thumbnail and return the thumbnail's binary content as a string.
+     *
+     * @param \BEdita\Core\Model\Entity\Stream $stream Stream entity instance.
+     * @param array $options Thumbnail options.
+     * @return string
+     */
+    protected function makeThumbnail(Stream $stream, array $options = [])
+    {
+        $source = (string)$stream->contents;
+
+        $thumbnail = $this->getGlideApi()
+            ->run($source, $options);
+
+        return $thumbnail;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getUrl(Stream $stream, array $options = [])
+    {
+        $path = $this->getFilename($stream, $options);
+
+        return FilesystemRegistry::getPublicUrl($path);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function generate(Stream $stream, array $options = [])
+    {
+        $path = $this->getFilename($stream, $options);
+
+        try {
+            $thumbnail = $this->makeThumbnail($stream, $options);
+
+            FilesystemRegistry::getMountManager()->put($path, $thumbnail);
+        } catch (NotReadableException $e) {
+            throw new InvalidStreamException(__('Unable to generate thumbnail for stream {0}', $stream->uuid), null, $e);
```

Comment:
should be `[$stream->uuid]`

Reply:
According to [`__()` function implementation](https://github.com/cakephp/cakephp/blob/master/src/I18n/functions.php#L22) it should be equivalent. Do you prefer to stick to this convention? I'm quite sure there are examples of both syntaxes in our code.

**Multiple arguments**: https://github.com/bedita/bedita/blob/4-cactus/plugins/BEdita/Core/src/Model/Action/RemoveAssociatedAction.php#L82
**Array**: https://github.com/bedita/bedita/blob/4-cactus/plugins/BEdita/Core/src/Model/Action/ChangeCredentialsAction.php#L95

Reply:
(BTW it should become `__d('bedita', ...)` anyways, at least)

Reply:
Yes, both should work. 
Scrutinizer is suggesting the array format. We may want to choose one form anyway... it's ok